app/src/test/java/net/osmtracker/activity/ButtonsPresetsTest.java

```diff
@@ -0,0 +1,310 @@
+        verify(mockActivity).setContentView(R.layout.buttons_presets);
+
+        // Check attributes are set
+        checkAttributesAfterInitialization(mockActivity);
+    }
+
+    @Test
+    @PrepareForTest({File.class, Preferences.class, Environment.class})
+    public void listLayoutsTest(){
+        for (int i = 0; i <= 1 ; i++) {
+            try {
+                doTestListLayouts(i);
+            }
+            catch (Exception e){
+                e.printStackTrace();
+                fail();
+            }
+        }
+    }
+
+
+
+
+
+    // TODO: make this test complete after refactoring the method,
+    //  now tests only a small part because couldn't be tested deeply
+    //  without refactoring or making it public
+    private void doTestListLayouts(int numberOfDownloadedLayouts) throws Exception {
+        ButtonsPresets mockActivity = mock(ButtonsPresets.class);
+
+        mockStatic(Environment.class); // avoid getExternal call
+
+        TextView mockEmpyText = mock(TextView.class);
+        when(mockActivity.findViewById(R.id.btnpre_empty)).thenReturn(mockEmpyText);
+
+        CheckBox mockDefaultcheckBox = mock(CheckBox.class);
+        when(mockDefaultcheckBox.getText()).thenReturn("foo");
+        when(mockActivity.findViewById(R.id.def_layout)).thenReturn(mockDefaultcheckBox);
+
+        injectMockHashtable(mockActivity, numberOfDownloadedLayouts);
+
+        callListLayouts(mockActivity, null);
+
+        int expectedVisibility = (numberOfDownloadedLayouts > 1) ? View.INVISIBLE : View.VISIBLE;
+        verify(mockEmpyText).setVisibility(expectedVisibility);
+
+
+
+
+    }
+
+    private void callListLayouts(ButtonsPresets mockActivity, LinearLayout mockRootLayout) {
+        try {
+            Method m = ButtonsPresets.class.getDeclaredMethod("listLayouts", LinearLayout.class);
+            m.setAccessible(true);
+            m.invoke(mockActivity, mockRootLayout);
+        }catch (Exception e){
+            e.printStackTrace();
+            fail();
+        }
+    }
+
+    private void mockDefaultSharedPreferences(Context context) {
+        SharedPreferences mockPrefs = mock(SharedPreferences.class);
+        mockStatic(PreferenceManager.class);
+        when(PreferenceManager.getDefaultSharedPreferences(context)).thenReturn(mockPrefs);
+    }
+
+    private void checkAttributesAfterInitialization(ButtonsPresets mockActivity) {
+        Hashtable hashtable = Whitebox.getInternalState(mockActivity.getClass(), "layoutsFileNames");
+        Object listener = Whitebox.getInternalState(mockActivity, "listener");
+        Object sharedPrefs = Whitebox.getInternalState(mockActivity, "prefs");
+        String storageDir = Whitebox.getInternalState(mockActivity.getClass(),"storageDir");
+
+        assertTrue(sharedPrefs instanceof SharedPreferences);
+        assertEquals(0, hashtable.size());
+        assertNotNull(listener);
+        assertEquals("//osmtracker", storageDir);
```

Comment:
why not use `File.separator + OSMTracker.Preferences.VAL_STORAGE_DIR` instead of `""//osmtracker"` ?

Reply:
Done!